src/backends/wasm_js.rs

```diff
@@ -1,158 +1,55 @@
-                    let sub_buf = buf.subarray(0, chunk_len);
-
-                    if crypto.get_random_values(&sub_buf).is_err() {
-                        return Err(Error::WEB_GET_RANDOM_VALUES);
-                    }
-
-                    // SAFETY: `sub_buf`'s length is the same length as `chunk`
-                    unsafe { sub_buf.raw_copy_to_ptr(chunk.as_mut_ptr().cast::<u8>()) };
-                }
-            }
-        };
-        Ok(())
-    })
-}
-
-fn getrandom_init() -> Result<RngSource, Error> {
     let global: Global = global().unchecked_into();
-
-    // Get the Web Crypto interface if we are in a browser, Web Worker, Deno,
-    // or another environment that supports the Web Cryptography API. This
-    // also allows for user-provided polyfills in unsupported environments.
     let crypto = global.crypto();
-    if crypto.is_object() {
-        let buf = Uint8Array::new_with_length(WEB_CRYPTO_BUFFER_SIZE.into());
-        Ok(RngSource::Web(crypto, buf))
-    } else if is_node(&global) {
-        // If module.require isn't a valid function, we are in an ES module.
-        let require_fn = Module::require_fn()
-            .and_then(JsCast::dyn_into::<Function>)
-            .map_err(|_| Error::NODE_ES_MODULE)?;
-        let n = require_fn
-            .call1(&global, &JsValue::from_str("crypto"))
-            .map_err(|_| Error::NODE_CRYPTO)?
-            .unchecked_into();
-        Ok(RngSource::Node(n))
-    } else {
-        Err(Error::WEB_CRYPTO)
-    }
-}
 
-// Taken from https://www.npmjs.com/package/browser-or-node
-fn is_node(global: &Global) -> bool {
-    let process = global.process();
-    if process.is_object() {
-        let versions = process.versions();
-        if versions.is_object() {
-            return versions.node().is_string();
+    // getRandomValues does not work with all types of WASM memory,
+    // so we initially write to browser memory to avoid exceptions.
+    let buf = Uint8Array::new_with_length(WEB_CRYPTO_BUFFER_SIZE.into());
+    for chunk in dest.chunks_mut(WEB_CRYPTO_BUFFER_SIZE.into()) {
+        let chunk_len: u32 = chunk
+            .len()
+            .try_into()
+            .expect("chunk length is bounded by WEB_CRYPTO_BUFFER_SIZE");
+        // The chunk can be smaller than buf's length, so we call to
+        // JS to create a smaller view of buf without allocation.
+        let sub_buf = buf.subarray(0, chunk_len);
+
+        if crypto.get_random_values(&sub_buf).is_err() {
+            return Err(Error::WEB_GET_RANDOM_VALUES);
         }
+
+        // SAFETY: `sub_buf`'s length is the same length as `chunk`
+        unsafe { sub_buf.raw_copy_to_ptr(chunk.as_mut_ptr().cast::<u8>()) };
     }
-    false
+    Ok(())
 }
 
 #[wasm_bindgen]
 extern "C" {
     // Return type of js_sys::global()
     type Global;
-
     // Web Crypto API: Crypto interface (https://www.w3.org/TR/WebCryptoAPI/)
     type WebCrypto;
```

Comment:
```suggestion
    type Crypto;
```
No need to differentiate it now. Would also align with what its actually called in JS.